### PR TITLE
Ported `org.lflang.tests` from Xtend to Java

### PR DESCRIPTION
**Pull request by lhstrh**
See https://github.com/lf-lang/lingua-franca/issues/838.

### CODE REVIEW

**Review by petervdonovan — APPROVED** [review]
I see nothing amiss here. (The change in `Progress.java` breaks the VS Code extension, of course, but it is necessary, and it will be addressed separately (#846).)

File: org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+         */
+        
+        model.eAllContents().forEachRemaining((obj) -> {
+            if (obj instanceof Parameter) {
+                Parameter parameter = (Parameter)obj;
+                if (parameter.getName() == "x") {
+                    var values = ASTUtils.initialValue(parameter, null);
+                    Assertions.assertEquals(values.get(0).getLiteral(), "1");
+                    
+                    values = ASTUtils.initialValue(parameter, list_a1);
```
Comment (oowekyala) [inline org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java:263]:
```suggestion
                    values = ASTUtils.initialValue(parameter, List.of(a1));
```
I would suggest removing all those list variables and replacing them with `List.of` like this. Alternatively you could also use `List.of` in the definition of those variables, which would save lines compared to LinkedList. I don't think using variables here nets any readability though

**Review by oowekyala — COMMENTED** [review]
Given the absence of text blocks these tests are surprinsingley readable! Cheers

File: org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java (hunk 1 above)
Comment (lhstrh) [reply org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java:263]: Thanks for the suggestion. Will do!

File: org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                    a2 = inst;
+                } else if (inst.getName().equals("b1")) {
+                    b1 = inst;
+                } else if (inst.getName().equals("b2")) {
+                    b2 = inst;
+                }
+            }
+        }
+        
+        return Map.of("a1", a1, "a2", a2, "b1", b1, "b2", b2);
```
Comment (oowekyala) [inline org.lflang.tests/src/org/lflang/tests/compiler/LinguaFrancaASTUtilsTest.java:180]:
```suggestion
       return asStream(model.eAllContents())
                .filter(obj -> obj instanceof Instantiation)
                .map(obj -> (Instantiation) obj)
                .collect(Collectors.toMap(Instantiation::getName, it -> it));
```

**Pull request merged**